app/assets/javascripts/components/articles/articles_handler.jsx

```diff
@@ -51,6 +55,12 @@ const ArticlesHandler = createReactClass({
      categories = <CategoryHandler course={this.props.course} current_user={this.props.current_user} />;
    }
 
+    this.props.addNotification({
```

Comment:
We only want to show a message if the limit has been reached.

Now that I think on it more, I also think it's important to retain a way for users to request the full list when they really want it. I think that could be done by:
* showing a notification only when the limit has been reached
* included a link or button in that notification that would request the full set of articles, with a warning that it might take a while.
* adding an optional `fetch_all` parameter one the server side of things, so that we use the same action for both limited and unlimited requests.